src/video.cpp

```diff
@@ -5,861 +5,908 @@
-					 * Using Shift but no other modifiers is an exception as this simply generates uppercase SDL_TEXTINPUT.
-					 * All keysyms that correspond to an ASCII character have the same integer value as this character;
-					 * all others are larger than the largest valid ASCII character (which is 0x7F).
-					 */
-					if (symbol_with_modifiers && event.key.keysym.sym <= 0x7F) {
-						std::string text;
-						text.push_back(event.key.keysym.sym);
-						return HandleKeyInput(WMKC_SYMBOL, this->modifier_state, text);
-					}
-
-					return false;
-			}
-			NOT_REACHED();
-
-		case SDL_KEYUP:
-			switch (event.key.keysym.sym) {
-				case SDLK_LCTRL:
-				case SDLK_RCTRL:
-					this->modifier_state &= ~WMKM_CTRL;
-					return true;
-
-				case SDLK_LALT:
-				case SDLK_RALT:
-				case SDLK_LGUI:
-				case SDLK_RGUI:
-					this->modifier_state &= ~WMKM_ALT;
-					return true;
-
-				case SDLK_LSHIFT:
-				case SDLK_RSHIFT:
-					this->modifier_state &= ~WMKM_SHIFT;
-					_game_control.action_test_mode = false;
-					return true;
-
-				default: return false;
-			}
+	if (!glfwInit()) error("Failed to initialize GLFW\n");
+
+	/* Create a window. */
+	glfwWindowHint(GLFW_SAMPLES, 4);  // 4x antialiasing.
+	glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);  // Require OpenGL 3.3 or higher.
+	glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 3);
+	glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GL_TRUE);
+	glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
+	this->width = 800;
+	this->height = 600;
+	this->mouse_x = this->width / 2;
+	this->mouse_y = this->height / 2;
+	this->mouse_dragging = MB_NONE;
 
-		case SDL_MOUSEMOTION:
-			UpdateMousePosition(event.button.x, event.button.y);
-			return false;
-
-		case SDL_MOUSEWHEEL:
-			_window_manager.MouseWheelEvent((event.wheel.y > 0) ? 1 : -1);
-			return false;
-
-		case SDL_MOUSEBUTTONUP:
-		case SDL_MOUSEBUTTONDOWN:
-			switch (event.button.button) {
-				case SDL_BUTTON_LEFT:
-					UpdateMousePosition(event.button.x, event.button.y);
-					_window_manager.MouseButtonEvent(MB_LEFT, (event.type == SDL_MOUSEBUTTONDOWN));
-					break;
-
-				case SDL_BUTTON_MIDDLE:
-					UpdateMousePosition(event.button.x, event.button.y);
-					_window_manager.MouseButtonEvent(MB_MIDDLE, (event.type == SDL_MOUSEBUTTONDOWN));
-					break;
-
-				case SDL_BUTTON_RIGHT:
-					UpdateMousePosition(event.button.x, event.button.y);
-					_window_manager.MouseButtonEvent(MB_RIGHT, (event.type == SDL_MOUSEBUTTONDOWN));
-					break;
-
-				default:
-					break;
-			}
-			return false;
+	std::string caption = "FreeRCT ";
+	caption += _freerct_revision;
+	this->window = glfwCreateWindow(this->width, this->height, caption.c_str(), NULL, NULL);
+	if (this->window == nullptr) {
+		glfwTerminate();
+		error("Failed to open GLFW window\n");
+	}
 
-		case SDL_USEREVENT:
-			return true;
+	glfwMakeContextCurrent(this->window);
+	if (glewInit() != GLEW_OK) error("Failed to initialize GLEW\n");
+	this->UpdateClip();
```

Comment:
`error` bails out too right?

Maybe insert an empty line between these to denote it's not one contiguous block?